src/frontend/src/pages/ProjectPage/pages/ContractPage/pages/ManagePersonnelPage/AddPersonnelSideSheet/index.tsx

```diff
@@ -0,0 +1,110 @@
+import * as classNames from 'classnames'
+import * as styles from './styles.less'
+import { useComponentDisplayClassNames, useCurrentContext } from '@equinor/fusion';
+import { generateRowTemplate, generateColumnTemplate } from './utils';
+import Header from './AddPersonnelFormHeader';
+import { useAppContext } from '../../../../../../../appContext';
+import { useContractContext } from '../../../../../../../contractContex';
+import AddPersonnelFormTextInput from './AddPersonnelFormTextInput';
+import useAddPersonnelForm from '../hooks/useAddPersonnelForm';
+
+
+type AddPersonnelToSideSheetProps = {
+  isOpen: boolean;
+  selectedPersonnel:Personnel[] | null
+  setIsOpen: (state:boolean) => void;
+}
+
+
+const AddPersonnelSideSheet: React.FC<AddPersonnelToSideSheetProps> = ({isOpen,setIsOpen,selectedPersonnel}) => {
+  const currentContext = useCurrentContext()
+  const currentContract = useContractContext()
+  const {
+    formState,
+    setFormState,
+    resetForm,
+    isFormValid,
+    isFormDirty,
+  } =  useAddPersonnelForm(selectedPersonnel)
+  const { apiClient } = useAppContext();
+  
+  const [saveInProgress,setSaveInProgress] = React.useState<Boolean>(false);
+  
+  const  savePersonnelChangesAsync = async () => {
+    const contractId = currentContract.contract?.id;
+
+
+    if(!currentContext?.id || !contractId )
+      return  
+    
+    setSaveInProgress(true);
+    await Promise.all(formState
+      .map(async person => await apiClient.updatePersonnelAsync(currentContext.id,contractId,person)))
+      .then(() => {
+        setSaveInProgress(false)
+        setIsOpen(false);
+      })
+      .catch(() => setSaveInProgress(false))  
+  }
+
+  const onChange = React.useCallback((changedPerson: Person)=>{
+    const updatedPersons = formState.map((p) => p.personnelId === changedPerson.personnelId ? changedPerson:p);
+    setFormState(updatedPersons);
+  },[formState,setFormState])
+
+  const headers = ["Name","Mail","Phone"]
+  const rowTemplate = generateRowTemplate(headers);
+  const columnTemplate = generateColumnTemplate(headers);
+  const containerClassNames = classNames(styles.container, useComponentDisplayClassNames(styles));
+  
+  return(
+    <ModalSideSheet
+      header="Add Person"
+      show={isOpen}
+      onClose={() => {
+        setIsOpen(false);
+      }}
+      safeClose
+      safeCloseTitle={`Close Add Person? Unsaved changes will be lost.`}
+      headerIcons={[
+        <Button key={"AddPerson"} outlined onClick = {()=> {setFormState([...formState,{personnelId:uuid(), name:"",phoneNumber:"",mail:"",jobTitle:""}])}} > + Add Person </Button>,
+        <Button disabled={!(isFormDirty && isFormValid)} key={"save"} outlined onClick = {()=> {savePersonnelChangesAsync()} }> Create </Button>
+      ]}
+      isResizable
+      minWidth={640}
+    >
+      {saveInProgress && <Spinner centered  size={100} title={"Saving Personnel"} />  }
+      {!saveInProgress && isOpen &&  <div className={containerClassNames}>
```

Comment:
Do we need to hide the entire table when saving? Wouldn't it be better for the user to just show a spinner in the save button or whatever while saving instead of switching the entire UI?